src/app.js

```diff
@@ -6,8 +6,225 @@ import _ from 'underscore';
+    const RES_FIELDS = ['trip_id', 'name', 'age', 'email'];
+    let reservationData = {}
+    RES_FIELDS.forEach((field) => {
+      const input = $(`#add-reservation input[name="${ field }"]`);
+      const val = input.val();
+      if (val != '') {
+        reservationData[field] = val;
+      }
+      input.val('');
+    })
+    const reservation = new Reservation(reservationData);
+    console.log(reservationData);
+    console.log(reservation);
+    if (!reservation.isValid()) {
+      handleValidationFailures(reservation.validationError);
+      return;
+    }
+    reservation.save({}, {
+      success: (model, response) => {
+        console.log('success');
+        reportStatus('success', 'Trip reserved!');
+        $('#add-reservation').hide();
+      },
+      error: (model, response) => {
+        console.log(response);
+        console.log(response.responseJSON["errors"]);
+        reportStatus('error', 'Not reserved!')
+        handleValidationFailures(response.responseJSON["errors"]);
+      },
+    });
+  };
+  //////////////////////////
+  // reservation handler ///
+  //////////////////////////
+  const reserveTemplateShow = function reserveTemplateShow(trip){
+    const reservationElement = $('#reservation-form');
+    reservationElement.html('');
+    const generatedHTMLreserve = reserveTemplate(trip.attributes);
+    console.log(trip.attributes);
+    reservationElement.append(generatedHTMLreserve);
+  };
+
+  $('#reserve').show();
+  $('#reserve').on('click', function(event) {
+    reserveTemplateShow(trip);
+    $('#reserve').hide();
+    $('#add-reservation').on('submit', function(event) {
+      event.preventDefault();
+      console.log('on submit');
+      reserveTrip(event);
+    });
+
+  });
+};
+
+// pulled out fetchTripDetails function above
+const fetchTripDetails = function fetchTripDetails(event) {
+  $('#add-reservation').hide();
+  let tripId = $(this).attr('data-id');
+  let trip = tripList.get(tripId)
+  trip.fetch({
+    success: () => {
+      showTripDetails(trip);
+    }
+  });
+};
+
+//////////////////////////
+// sorting ///////////////
+//////////////////////////
+const render = function render(tripList) {
+  const tripListElement = $('#trip-list');
+  tripListElement.empty();
+
+  tripList.forEach((trip) => {
+    const generatedHTML = tripTemplate(trip.attributes);
+    tripListElement.append($(generatedHTML));
+
+  });
+  $('th.sort').removeClass('current-sort-field');
+  $(`th.sort.${ tripList.comparator }`).addClass('current-sort-field');
+  $('.trip').on('click', fetchTripDetails);
+};
+
+//////////////////////////
+// Add a Trip  ///////////
+//////////////////////////
+
+const readFormData = function readFormData() {
+  const tripData = { id: null };
+  TRIP_FIELDS.forEach((field) => {
+    const inputElement = $(`#add-trip-form input[name=${ field }]`);
+    const value = inputElement.val();
+    if (value != '') {
+      tripData[field] = value;
+    }
+    inputElement.val('');
```

Comment:
Also clearing the values immediately means that the user will have to re-enter all the data on validation errors.  Instead consider clearing the fields on a successful save.